src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortEngine.java

```diff
@@ -450,12 +454,15 @@ private void finalizeCurrentVariant(final List<VariantContext> unmergedCalls,
         final VariantContext finalVC = removeAnnotations(filteredVC);
 
         if ( finalVC != null ) {
-            vcfWriter.add(finalVC);
+            // If XLfiltered is true, than the filtered sites should not be added
+            if (!XLfiltered || finalVC.isNotFiltered()) {
```

Comment:
this made my brain hurt :D.  What do you think of `! (XLfiltered && finalVC.isFiltered)`.  Meh, ok I don't love that either 🤷

Reply:
hahaha Andrea listened to me repeat this about 30 times before it was correct. I'm not sure of the best option, but I can comment it more clearly for sure

Reply:
is it any clearer if I just swap them?
if (finalVC.isNotFiltered() || !XLfiltered ) { foo }